### PR TITLE
Resolve updated codecheck issues

### PR DESCRIPTION
**Pull request by adlarkin**
This PR is meant to resolve new codecheck issues introduced by https://github.com/ignitionrobotics/ign-cmake/pull/117.

From my understanding, the plan is to forward port the fixes here to all other ignition versions once https://github.com/ignitionrobotics/ign-cmake/pull/117 is merged.

Signed-off-by: Ashton Larkin <ashton@openrobotics.org>

### CODE REVIEW

File: log/src/cmd/LogCommandAPI.cc [hunk 1]
```diff
@@ -15,17 +15,19 @@
...
 #include <regex>
+#include <string>
 
 #include <ignition/transport/log/Export.hh>
 #include <ignition/transport/log/Playback.hh>
 #include <ignition/transport/log/Recorder.hh>
 #include <ignition/transport/Node.hh>
 #include <ignition/transport/NodeOptions.hh>
 #include "../Console.hh"
-#include "LogCommandAPI.hh"
```
Comment (chapulina) [inline log/src/cmd/LogCommandAPI.cc:28]: Just curious, does the new linter require that only the header for this source file be up top?

**Review by chapulina — APPROVED** [review]
Just a question about the moved internal headers.

File: log/src/cmd/LogCommandAPI.cc (hunk 1 above)
Comment (adlarkin) [reply log/src/cmd/LogCommandAPI.cc:28]: It does not - I moved it to follow the [google style guide](https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes), which states:

> Include headers in the following order: Related header, C system headers, C++ standard library headers, other libraries' headers, your project's headers.

(note the related header being first) should I leave it the way it was? I made this change (making the related header being the first thing included) in other files I modified for codecheck as well.

Comment (chapulina) [reply log/src/cmd/LogCommandAPI.cc:28]: I think it's fine, it looks like the local codechecker is happy here. It isn't happy in other libraries though, the `tools/code_check.sh` script may be slightly different for those libraries.

Comment (adlarkin) [reply log/src/cmd/LogCommandAPI.cc:28]: I made a [comment](https://github.com/ignitionrobotics/ign-gui/pull/144#discussion_r517073234) about the other library where this is an issue - I think the other library actually had an issue with the QT header, and that there isn't a difference with the `tools/code_check.sh` script.

**Pull request merged**